src/components/CalendarEventForMonthView.tsx

```diff
@@ -12,36 +13,73 @@ interface CalendarEventProps<T> {
   onPressEvent?: (event: ICalendarEvent<T>) => void
   eventCellStyle?: EventCellStyle<T>
   renderEvent?: EventRenderer<T>
+  date: dayjs.Dayjs
+  dayOfTheWeek: number
 }
 
 function _CalendarEventForMonthView<T>({
   event,
   onPressEvent,
   eventCellStyle,
   renderEvent,
+  date,
+  dayOfTheWeek,
 }: CalendarEventProps<T>) {
   const theme = useTheme()
+  const { width } = Dimensions.get('window')
+  const dayWidth = width / 7
+
+  // adding + 1 because durations start at 0
+  const eventDuration = dayjs.duration(dayjs(event.end).diff(dayjs(event.start))).days() + 1
+  const eventDaysLeft = dayjs.duration(dayjs(event.end).diff(date)).days() + 1
+  const weekDaysLeft = 7 - dayOfTheWeek
+  const isMultipleDays = eventDuration > 1
+  // This is to determine how many days from the event to show during a week
+  const eventWeekDuration =
+    eventDuration > weekDaysLeft
+      ? weekDaysLeft
+      : dayOfTheWeek === 0 && eventDaysLeft < eventDuration
+      ? eventDaysLeft
+      : eventDuration
+  const isMultipleDaysStart =
+    isMultipleDays &&
+    (date.isSame(event.start, 'day') || (dayOfTheWeek === 0 && date.isAfter(event.start)))
+  // - 6 to take in account the padding
+  const eventWidth = dayWidth * eventWeekDuration - 6
```

Comment:
I think this part should be

- moved to `utils.ts` 
- ideally unit tested
- memoized

Reply:
I have moved the code to utils and memoized it. I have also added unit testing but since I return a width in my object, I'm not sure how to test that result.